certbot-nginx/certbot_nginx/tls_sni_01.py

```diff
@@ -141,7 +140,11 @@ def _make_server_block(self, achall, addrs):
         document_root = os.path.join(
             self.configurator.config.work_dir, "tls_sni_01_page")
 
-        block = [['listen', ' ', str(addr)] for addr in addrs]
+        no_default_addrs = [copy.deepcopy(addr) for addr in addrs]
```

Comment:
I agree it's a bit unpythonic. Googling around, I couldn't find a common pattern used for controlling the output of `str`. This seems a bit silly to me as it doesn't seem that unusual to need display an object in different ways depending on the context.

I think the benefits of writing a `<insert_alternative_string_method_name_here>` method over the current solution is:

* Less code
* Faster
* Can easily be reused

With all that said, I don't think it's clear what's best here and if you want to use your current approach...

Looks like you coded it up before I could respond :smile: